.github/workflows/test.yml

```diff
@@ -205,6 +206,86 @@ jobs:
           file: ./coverage.txt
         if: "env.GIT_DIFF != ''"
 
+  test-race-1:
+    runs-on: ubuntu-latest
+    needs: split-test-files
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v2
+      - uses: technote-space/get-diff-action@v3
+        id: git_diff
+        with:
+          SUFFIX_FILTER: |
+            .go
+            .mod
+            .sum
+      - uses: actions/download-artifact@v2
+        with:
+          name: "${{ github.sha }}-aa"
+        if: "env.GIT_DIFF != ''"
+      - name: Run tests with race detector
+        run: cat xaa.txt | xargs go test -mod=readonly -timeout 15m -race -tags='cgo ledger test_ledger_mock'
+
+  test-race-2:
+    runs-on: ubuntu-latest
+    needs: split-test-files
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v2
+      - uses: technote-space/get-diff-action@v3
+        id: git_diff
+        with:
+          SUFFIX_FILTER: |
+            .go
+            .mod
+            .sum
+      - uses: actions/download-artifact@v2
+        with:
+          name: "${{ github.sha }}-ab"
+        if: "env.GIT_DIFF != ''"
+      - name: Run tests with race detector
+        run: cat xab.txt | xargs go test -mod=readonly -timeout 15m -race -tags='cgo ledger test_ledger_mock'
+
+  test-race-3:
+    runs-on: ubuntu-latest
+    needs: split-test-files
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v2
+      - uses: technote-space/get-diff-action@v3
+        id: git_diff
+        with:
+          SUFFIX_FILTER: |
+            .go
+            .mod
+            .sum
+      - uses: actions/download-artifact@v2
+        with:
+          name: "${{ github.sha }}-ac"
+        if: "env.GIT_DIFF != ''"
+      - name: Run tests with race detector
+        run: cat xac.txt | xargs go test -mod=readonly -timeout 15m -race -tags='cgo ledger test_ledger_mock'
+
+  test-race-4:
+    runs-on: ubuntu-latest
+    needs: split-test-files
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v2
+      - uses: technote-space/get-diff-action@v3
+        id: git_diff
+        with:
+          SUFFIX_FILTER: |
+            .go
+            .mod
+            .sum
+      - uses: actions/download-artifact@v2
+        with:
+          name: "${{ github.sha }}-ad"
+        if: "env.GIT_DIFF != ''"
+      - name: Run tests with race detector
+        run: cat xad.txt | xargs go test -mod=readonly -timeout 15m -race -tags='cgo ledger test_ledger_mock'
```

Comment:
Why not running all tests in the same action as an additional step?

This would save many CPU cycles (not sure if we pay for that).

Reply:
No, we don't pay for that. And we split test runs to run them in parallel